### PR TITLE
43.login to profile functionality

### LINKED ISSUE #43: Login to Profile Functionality
User Requirement: 03.01.01

Rationale: As a participant, I want to be able to log into my profile with my username and password, so I can persistently use the app

Description: Refine the login page to be able to create a user profile such that if a user enters a unique ID that does not already exist, the account gets created in the database and sends them to the myVibes activity. If a user enters an existing account and their password,  it will log them in by verifying the account in the database. Otherwise display a prompt that indicates a user has entered invalid credentials or the account does not exist (this should be a Toast).

Story Points: 13

#16

### PR DESCRIPTION
**Pull request by jacobwong98**
[//]: # (These are comments and are used for reference and will not show up in the PR)

Description: 

- Implemented Login Functionality (Login with email and password)
- Implemented Create Account Functionality (Create account with username, email and password)
- Save the user's email and username in Firestore under Users collection
- Verify unique username with Firestore Users collection
- Redirect user to MyVibesActivity if they are logged in

[//]: # (Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change)

[//]: # (automatically closes given issue number)
Closes #43  

Proof of Concept

image
image

[//]: # (Attach Image of feature added or fix applied and working)
[//]: # (Please provide images of tests passing)

Checklist

[//]: # (Add an x between square brackets to mark item as done)

- [x] Follows style guidelines and is **properly commented**
- [x] Code changes generate no new warnings
- [ ] Testing (Unit and/or Intent)
- [x] Documentation (UML, Wiki, etc.)

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -26,9 +26,31 @@
...
+        android:padding="25dp"
         android:textSize="48sp" />
 
+<!--    <EditText-->
+<!--        android:id="@+id/lat_edit_text"-->
+<!--        android:layout_width="match_parent"-->
+<!--        android:layout_height="wrap_content"-->
+<!--        android:autofillHints="false"-->
+<!--        android:hint="Latitude (-90 to 90)"-->
+<!--        android:inputType="numberDecimal|numberSigned" />-->
```
Comment (raysarinas) [inline app/src/main/res/layout/activity_main.xml]: why is this block of code commented out? lol please remove if not necessary

File: app/src/main/java/com/example/vybe/MainActivity.java [hunk 2]
```diff
@@ -22,17 +42,88 @@ protected void onCreate(Bundle savedInstanceState) {
...
+     * @return Returns true if the field is empty, false it is filled
+     */
+    public static boolean isEmpty(EditText et){
+        return et.getText().toString().trim().length() == 0;
+    }
+
+    /**
+     * Helper function to get trimmed string from edit text field
+     * @param et
+     *      EditText that we want to get the string from
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/MainActivity.java]: not 2 be picky but can u just rename this to `edittext`

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (jacobwong98) [reply app/src/main/res/layout/activity_main.xml]: Will remove

File: app/src/main/java/com/example/vybe/MainActivity.java (hunk 2 above)
Comment (jacobwong98) [reply app/src/main/java/com/example/vybe/MainActivity.java]: sure thing scrum master

**Comment by jacobwong98** [comment]
> 
> 
> Looks really good! 💯 Especially the comments 🥇
> 
> BUT that being said, this is going to be annoying, but will help in the long run: please refactor any "user" related information and put them into a model "User" class - like how we have a "Vibe" to be a container for an emoticon and color, I believe a User class should contain the username, email, and probably like a list of Vibe Events (not entirely sure on that). Probably best to look @ how Pranav did the UML. Once that is done, me or @TheBroMoe can write Unit Tests for this User class.
> 
> Also I do think once all appropriate changes are made, we will need to hand-off this issue to @gatilogo to fix the XML

Should this be a new issue or refactor this now?

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (jacobwong98) [reply app/src/main/res/layout/activity_main.xml]: Done

**Comment by raysarinas** [comment]
> > Looks really good! 💯 Especially the comments 🥇
> > BUT that being said, this is going to be annoying, but will help in the long run: please refactor any "user" related information and put them into a model "User" class - like how we have a "Vibe" to be a container for an emoticon and color, I believe a User class should contain the username, email, and probably like a list of Vibe Events (not entirely sure on that). Probably best to look @ how Pranav did the UML. Once that is done, me or @TheBroMoe can write Unit Tests for this User class.
> > Also I do think once all appropriate changes are made, we will need to hand-off this issue to @gatilogo to fix the XML
> 
> Should this be a new issue or refactor this now?

I think better now because it might get worse if we keep putting it off lol kind of like how we are now stuck with a big refactoring issue

**Comment by jacobwong98** [comment]
> 
> 
> > > Looks really good! 💯 Especially the comments 🥇
> > > BUT that being said, this is going to be annoying, but will help in the long run: please refactor any "user" related information and put them into a model "User" class - like how we have a "Vibe" to be a container for an emoticon and color, I believe a User class should contain the username, email, and probably like a list of Vibe Events (not entirely sure on that). Probably best to look @ how Pranav did the UML. Once that is done, me or @TheBroMoe can write Unit Tests for this User class.
> > > Also I do think once all appropriate changes are made, we will need to hand-off this issue to @gatilogo to fix the XML
> > 
> > 
> > Should this be a new issue or refactor this now?
> 
> I think better now because it might get worse if we keep putting it off lol kind of like how we are now stuck with a big refactoring issue

@raysarinas please review these changes thanks!

**Comment by jacobwong98** [comment]
> 
> 
> sensing some attitude but yes I approve I guess 👍
> 
> I'll write some unit tests and then we can merge

Tests looks good. Github won't let me confirm the review

**Pull request merged**